src/components/Metrics/IstioMetrics.tsx

```diff
@@ -161,6 +161,13 @@ class IstioMetrics extends React.Component<IstioMetricsProps, MetricsState> {
     this.fetchMetrics();
   };
 
+  onClickDataPoint = (_, datum: VCDataPoint) => {
+    if ('traceId' in datum) {
+      HistoryManager.setParam(URLParam.TAB, 'traces');
+      HistoryManager.setParam(URLParam.JAEGER_TRACE_ID, datum.traceId);
```

Comment:
Here this works fine when we look at metrics from service details, but we can also see span from app page, so it would fail because it's going to find trace tab in app details, which doesn't exist.
I think it should always redirect to service details/traces, so we need to rebuild the full url, something like:

`/namespaces/${this.props.namespace}/services/${this.props.object}?${URLParam.TAB}=traces&${URLParam.JAEGER_TRACE_ID}=${datum.traceId}`

Reply:
Also the same behaviour should be done in CustomMetrics.tsx for runtimes

Reply:
Wow I didn't think this cases...so I need to do more changes thanks @jotak